pkg/config/remote/service/websocket_echo.go

```diff
@@ -0,0 +1,98 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/remote-config/rcwebsocket"
+	"github.com/DataDog/datadog-agent/pkg/config/remote/api"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// WebSocketTestActor periodically calls rcwebsocket.RunWebSocketTest() in a
+// background task.
+type WebSocketTestActor struct {
+	client *api.HTTPClient
+	// Callback to run the test.
+	fn func(context.Context, *api.HTTPClient)
+
+	stopCh chan struct{} // nil until Start() called
+}
+
+// NewWebSocketTestActor constructs a NewWebSocketTestActor that uses client to
+// obtain a WebSocket connection to the RC backend.
+func NewWebSocketTestActor(client *api.HTTPClient) *WebSocketTestActor {
+	return &WebSocketTestActor{
+		client: client,
+		fn:     rcwebsocket.RunEchoTest,
+	}
+}
+
+// Start runs this actor, spawning a background task to execute the test
+// periodically.
+//
+// This method is not concurrency safe, and panics if Start() has previously
+// been called.
+func (s *WebSocketTestActor) Start() {
+	if s.stopCh != nil {
+		panic("attempt to start WebSocketTestActor more than once")
+	}
+
+	s.stopCh = make(chan struct{})
+
+	go func() {
+		s.run()
+	}()
+}
```

Comment:
I make the assumption that [the `CoreAgentService.Start()` method](https://github.com/DataDog/datadog-agent/blob/5624e49c09e450b27333fbb0b6969a763d951ac1/pkg/config/remote/service/service.go/#L542) is called once (and therefore concurrently).

I also assume that [the `CoreAgentService.Stop()` method](https://github.com/DataDog/datadog-agent/blob/5624e49c09e450b27333fbb0b6969a763d951ac1/pkg/config/remote/service/service.go/#L639) _can_ be called more than once (at least one test does) but never concurrently.

Are these assumptions correct?